lib/utils.ak

```diff
@@ -108,6 +119,28 @@ pub fn get_own_validator_inputs(
   )
 }
 
+pub fn get_other_validator_inputs(
+  ctx: ScriptContext,
+  output_reference: OutputReference,
+) -> List<Input> {
+  let script_own_address = get_own_validator_address(ctx, output_reference)
+
+  // Check if there is at least one another SC with datum and address is not it's own
+  list.filter(
+    ctx.transaction.inputs,
+    fn(input) {
+      has_datum(input.output.datum) && input.output.address != script_own_address
```

Comment:
This seems non-obvious to me. I'd check the credential of `input.output.address` instead, it should be `ScriptCredential`.
https://aiken-lang.github.io/stdlib/aiken/transaction/credential.html#Credential